lms/djangoapps/courseware/tests/test_masquerade.py

```diff
@@ -281,6 +282,26 @@ def get_progress_detail(self):
         return json.loads(self.look_at_question(self.problem_display_name).content)['progress_detail']
 
     @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
+    def test_masquerade_as_specific_user_on_self_paced(self):
+        """
+        Test masquerading as a specific user for course info page when self paced configuration
+        "enable_course_home_improvements" flag is set
+
+        Login as a staff user and visit course info page.
+        set masquerade to view same page as a specific student and revisit the course info page.
+        """
+        # Log in as staff, and check we can see the info page.
+        self.login_staff()
+        content = self.get_course_info_page().content
+        self.assertIn("OOGIE BLOOGIE", content)
+
+        # Masquerade as the student,enable the self paced configuration, and check we can see the info page.
+        SelfPacedConfiguration(enable_course_home_improvements=True).save()
+        self.update_masquerade(role='student', user_name=self.student_user.username)
+        content = self.get_course_info_page().content
```

Comment:
I think we should get the complete response here. And then check if we are getting response status 200 as well.

Just like this : 
```
response = self.get_course_info_page()
self.assertEqual(response.status_code, 200)
content = response.content
```